app/controllers/requests_controller.rb

```diff
@@ -115,12 +115,25 @@ def delegated_new_request_path(request, url_params = nil)
   end
   helper_method :delegated_new_request_path
 
+  # This takes the checkbox-based hash parameter (which is the format that we allow in our strong params).
+  # We persist an array of checkboxes so rails wants an array passed in.
   def modify_item_selector_checkboxes
     return unless create_params && create_params[:barcodes].is_a?(ActionController::Parameters)
 
     create_params[:barcodes] = create_params[:barcodes].select { |_, checked| checked == '1' }.keys
   end
 
+  # Radio buttons get sent in as an array (radios are used when only a single item is allowed from a mutli-volume set)
+  # This will modify the barcodes array to be a hash that mimics checkbox elements (for strong params purposes).
+  # This worked pre-rails 5 but now we're restricted to only passing one parameter format.
+  # modify_item_selector_checkboxes will change this back into an array (which is dumb) to be persisted, but this is
+  # seemingly the only way to get Strong Params + AR Serializer + Checkbox/Radio agnostic form element to work.
+  def modify_item_selector_radio_buttons
+    return unless params[:request][:barcodes].is_a?(Array) && params[:request][:barcodes].one?
+
+    create_params[:barcodes] = params[:request][:barcodes].map { |barcode| [barcode, '1'] }.to_h
```

Comment:
I think I'm confused what the contract is.. we're turning radio buttons into a hash, and we're turning checkboxes into an array?